pinot-common/src/main/java/org/apache/pinot/common/utils/DataSchema.java

```diff
@@ -353,6 +353,9 @@ public Serializable convert(Object value) {
         case BOOLEAN:
           return ((Number) value).intValue() == 1;
         case TIMESTAMP:
+          if (value instanceof Timestamp) {
```

Comment:
Does this still apply? Ideally we should always have internal format value. If we break that contract, `BOOLEAN` will also break